runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcGetWorkerMetadataStreamTest.java

```diff
@@ -0,0 +1,336 @@
+          .build();
+  private static final String FAKE_SERVER_NAME = "Fake server for GrpcGetWorkerMetadataStreamTest";
+  @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+  private final MutableHandlerRegistry serviceRegistry = new MutableHandlerRegistry();
+  private final Set<AbstractWindmillStream<?, ?>> streamRegistry = new HashSet<>();
+  private ManagedChannel inProcessChannel;
+  private GrpcGetWorkerMetadataStream stream;
+
+  private GrpcGetWorkerMetadataStream getWorkerMetadataTestStream(
+      GetWorkerMetadataTestStub getWorkerMetadataTestStub,
+      int metadataVersion,
+      Consumer<WindmillEndpoints> endpointsConsumer) {
+    serviceRegistry.addService(getWorkerMetadataTestStub);
+    return GrpcGetWorkerMetadataStream.create(
+        responseObserver ->
+            CloudWindmillServiceV1Alpha1Grpc.newStub(inProcessChannel)
+                .getWorkerMetadataStream(responseObserver),
+        FluentBackoff.DEFAULT.backoff(),
+        StreamObserverFactory.direct(DEFAULT_STREAM_RPC_DEADLINE_SECONDS * 2, 1),
+        streamRegistry,
+        1, // logEveryNStreamFailures
+        TEST_JOB_HEADER,
+        metadataVersion,
+        new ThrottleTimer(),
+        endpointsConsumer);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    Server server =
+        InProcessServerBuilder.forName(FAKE_SERVER_NAME)
+            .fallbackHandlerRegistry(serviceRegistry)
+            .directExecutor()
+            .build()
+            .start();
+
+    inProcessChannel =
+        grpcCleanup.register(
+            InProcessChannelBuilder.forName(FAKE_SERVER_NAME).directExecutor().build());
+    grpcCleanup.register(server);
+    grpcCleanup.register(inProcessChannel);
+    GLOBAL_DATA_ENDPOINTS.put(
+        "global_data",
+        WorkerMetadataResponse.Endpoint.newBuilder()
+            .setDirectEndpoint(IPV6_ADDRESS_1)
+            .setWorkerToken("worker_token")
+            .build());
+  }
+
+  @After
+  public void cleanUp() {
+    inProcessChannel.shutdownNow();
+  }
+
+  @Test
+  public void testGetWorkerMetadata() {
+    WorkerMetadataResponse mockResponse =
+        WorkerMetadataResponse.newBuilder()
+            .setMetadataVersion(1)
+            .addAllWorkEndpoints(DIRECT_PATH_ENDPOINTS)
+            .putAllGlobalDataEndpoints(GLOBAL_DATA_ENDPOINTS)
+            .build();
+    TestWindmillEndpointsConsumer testWindmillEndpointsConsumer =
+        new TestWindmillEndpointsConsumer();
+    GetWorkerMetadataTestStub testStub =
+        new GetWorkerMetadataTestStub(new TestGetWorkMetadataRequestObserver());
+    int metadataVersion = -1;
+    stream = getWorkerMetadataTestStream(testStub, metadataVersion, testWindmillEndpointsConsumer);
+    testStub.injectWorkerMetadata(mockResponse);
+
+    assertEquals(
+        GLOBAL_DATA_ENDPOINTS.size(), testWindmillEndpointsConsumer.globalDataEndpoints.size());
+    testWindmillEndpointsConsumer.globalDataEndpoints.forEach(
```

Comment:
are there matchers you could use instead? This is where having someone more familiar with Java could help review.

Reply:
found them adding the dependency for the gradle file.  I will update the tests to reflect and then submit

using google.Truth